crates/shim/src/lib.rs

```diff
@@ -180,29 +198,38 @@ pub trait Shim {
 }
 
 /// Shim entry point that must be invoked from `main`.
-pub fn run<T>(runtime_id: &str)
+pub fn run<T>(runtime_id: &str, opts: Vec<fn(&mut Config)>)
```

Comment:
Let's just pass `opts: Option<Config>` here
and below `opts.unwrap_or_default()`

Reply:
`Option<Config>`? A single config or `None`, rather than `fn(&mut Config)` vectors:
https://github.com/containerd/containerd/blob/main/runtime/v2/shim/shim.go#L181

Reply:
Done, quite simple